pkgs/applications/science/logic/ls4/default.nix

```diff
@@ -0,0 +1,36 @@
+{ zlib, gcc, stdenv, fetchFromGitHub, overrideCC }:
+
+(overrideCC stdenv gcc).mkDerivation rec {
+  pname = "ls4";
+
+  version = "unstable-2019-07-31";
+
+  buildInputs = [ zlib gcc ];
+
+  preBuild = ''
+  cd core
+  # These object files were committed into the git repo, perhaps accidentally
+  rm -f aiger.o
+  rm -f aiger.o_32
+  gcc -g -O3 -c aiger.c
```

Comment:
Why not use make as per documentation?

Reply:
This is an error in the upstream source repo. The Makefile it includes doesn't actually re-build these object files and just expects them to exist, but the pre-existing object files aren't built correctly for all architectures.

I'm guessing it's because of convenience; the rest of the Makefile just has to call the g++ compiler instead of the gcc compiler here.